internal/controllers/utils/oauth.go

```diff
@@ -71,12 +77,25 @@ func SetupOAuthClient(ctx context.Context, config *OAuthClientConfig) (*http.Cli
 			AuthStyle:      oauth2.AuthStyleInParams,
 		}
 
-		ctx = context.WithValue(ctx, oauth2.HTTPClient, c)
+		ctx = context.WithValue(ctx, oauth2.HTTPClient, baseClient)
+		oauthClient := oauthConfig.Client(ctx)
+
+		// Verify token acquisition works
+		if src, ok := oauthClient.Transport.(*oauth2.Transport); ok {
+			// Force token acquisition now to verify it works
+			_, err := src.Source.Token()
+			if err != nil {
+				return nil, fmt.Errorf("failed to acquire initial OAuth token: %w", err)
+			}
+			slog.Info("Successfully acquired initial OAuth token")
+		}
```

Comment:
I don't think that failing here is a reasonable approach.  The OAuth server may be momentarily unreachable... i'd rather defer this type of failure when the token is actually needed.

Reply:
removed